src/main/java/edu/ie3/datamodel/io/source/sql/SqlIdCoordinateSource.java

```diff
@@ -0,0 +1,315 @@
+    List<CoordinateValue> values = executeQuery(queryForPoint, ps -> ps.setInt(1, id));
+
+    return Optional.of(values.get(0).coordinate);
+  }
+
+  @Override
+  public Collection<Point> getCoordinates(int... ids) {
+    Object[] idSet = Arrays.asList(ids, ids.length).toArray();
+
+    List<CoordinateValue> values =
+        executeQuery(
+            queryForPoint,
+            ps -> {
+              Array sqlArray = ps.getConnection().createArrayOf("integer", idSet);
+              ps.setArray(1, sqlArray);
+            });
+
+    List<Point> points = new ArrayList<>();
+
+    for (CoordinateValue value : values) {
+      points.add(value.coordinate);
+    }
+
+    return points;
+  }
+
+  @Override
+  public Optional<Integer> getId(Point coordinate) {
+    double latitude = coordinate.getY();
+    double longitude = coordinate.getX();
+
+    List<CoordinateValue> values =
+        executeQuery(
+            queryForId,
+            ps -> {
+              ps.setDouble(1, latitude);
+              ps.setDouble(2, longitude);
+            });
+
+    return Optional.of(values.get(0).id);
+  }
+
+  @Override
+  public Collection<Point> getAllCoordinates() {
+    List<CoordinateValue> values = executeQuery(basicQuery, ps -> ps.setString(1, ";"));
+
+    ArrayList<Point> points = new ArrayList<>();
+
+    for (CoordinateValue value : values) {
+      points.add(value.coordinate);
+    }
+
+    return points;
+  }
+
+  @Override
+  public List<CoordinateDistance> getNearestCoordinates(Point coordinate, int n) {
+    double[] xyDeltas = calculateXYDelta(coordinate);
+
+    double longitude = coordinate.getX();
+    double latitude = coordinate.getY();
+
+    List<CoordinateValue> values =
+        executeQuery(
+            queryForBoundingBox,
+            ps -> {
+              ps.setDouble(1, longitude - xyDeltas[0]);
+              ps.setDouble(2, longitude + xyDeltas[0]);
+              ps.setDouble(3, latitude - xyDeltas[1]);
+              ps.setDouble(4, latitude + xyDeltas[1]);
+            });
+
+    ArrayList<Point> points = new ArrayList<>();
+
+    for (CoordinateValue value : values) {
+      points.add(value.coordinate);
+    }
+
+    List<CoordinateDistance> distances = getNearestCoordinates(coordinate, 2 * n, points);
+
+    return checkForBoundingBox(coordinate, distances, n);
+  }
+
+  @Override
+  public List<CoordinateDistance> getNearestCoordinates(
+      Point coordinate, int n, Collection<Point> coordinates) {
+    return IdCoordinateSource.super.getNearestCoordinates(coordinate, n, coordinates);
+  }
+
+  /**
+   * Method for evaluating the found points and returning the n corner points of the bounding box.
+   *
+   * @param coordinate at the center of the bounding box
+   * @param distances list of found points with their distances
+   * @param numberOfPoints that should be returned
+   * @return list of distances
+   */
+  private List<CoordinateDistance> checkForBoundingBox(
```

Comment:
Could you find a more fitting name for this method? Maybe something like `extractBoundingBox` or `restrictToBoundingBox`...